commons/src/main/java/com/powsybl/commons/datasource/TarArchiveDataSource.java

```diff
@@ -0,0 +1,330 @@
+        this(directory, tarFileName, baseName, mainExtension, compressionFormat, null);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, String mainExtension, CompressionFormat compressionFormat, DataSourceObserver observer) {
+        this(directory,
+            baseName + (mainExtension == null || mainExtension.isEmpty() ? "" : "." + mainExtension) + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, mainExtension, compressionFormat, observer);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, String mainExtension, CompressionFormat compressionFormat) {
+        this(directory,
+            baseName + (mainExtension == null || mainExtension.isEmpty() ? "" : "." + mainExtension) + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, mainExtension, compressionFormat, null);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, CompressionFormat compressionFormat, DataSourceObserver observer) {
+        this(directory,
+            baseName + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, null, compressionFormat, observer);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName, CompressionFormat compressionFormat) {
+        this(directory,
+            baseName + ".tar" + (compressionFormat == null ? "" : "." + compressionFormat.getExtension()),
+            baseName, null, compressionFormat, null);
+    }
+
+    public TarArchiveDataSource(Path directory, String baseName) {
+        this(directory,
+            baseName + ".tar",
+            baseName, null, null, null);
+    }
+
+    public TarArchiveDataSource(Path tarFile) {
+        this(tarFile.getParent(), (new FileInformation(tarFile.getFileName().toString(), false)).getBaseName(),
+            (new FileInformation(tarFile.getFileName().toString(), false)).getMainExtension(),
+            (new FileInformation(tarFile.getFileName().toString(), false)).getCompressionFormat());
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Files are here located in the archive.</p>
+     */
+    @Override
+    public Set<String> listNames(String regex) throws IOException {
+        // Initialize variables
+        Pattern p = Pattern.compile(regex);
+        Set<String> names = new HashSet<>();
+        Path tarFilePath = getArchiveFilePath();
+
+        // Explore the archive
+        try (BufferedInputStream inputStream = new BufferedInputStream(Files.newInputStream(tarFilePath));
+             InputStream cis = getCompressedInputStream(inputStream, compressionFormat);
+             TarArchiveInputStream tar = new TarArchiveInputStream(cis)) {
+            ArchiveEntry entry;
+            while ((entry = tar.getNextEntry()) != null) {
+                // Consider only files located at the root of the archive
```

Comment:
Maybe as a first version do the same as zip ? only filter on the matcher. (in the zip code the comment is wrong). Then in a separate PR we can decide if we implement this behavior for both ?

Reply:
https://github.com/powsybl/powsybl-core/blob/feeb6bc5d68427419939c40b7bd53d554003fe0c/commons/src/main/java/com/powsybl/commons/datasource/ZipFileDataSource.java#L212

but

https://github.com/powsybl/powsybl-core/blob/feeb6bc5d68427419939c40b7bd53d554003fe0c/commons/src/main/java/com/powsybl/commons/datasource/ZipFileDataSource.java#L222